commons/ava_networks/high_level_gecko_client.go

```diff
@@ -305,7 +308,7 @@ func (highLevelGeckoClient HighLevelGeckoClient) waitForXchainTransactionAccepta
 	if err != nil {
 		return stacktrace.Propagate(err,"Failed to get status.")
 	}
-	for status != TRANSACTION_ACCEPTED_STATUS {
+	for i := 0; i < highLevelGeckoClient.networkAcceptanceTimeoutInSeconds && status != TRANSACTION_ACCEPTED_STATUS; i++ {
```

Comment:
Ooooo really not a fan of doing it this way; it implicitly relies on the `time.Sleep` being a second here so if an innocent developer changes the `time.Sleep(time.Second)` to 2 seconds then this will break. That would be really subtle and nasty and spooky-action-at-a-distance.

The right way IMO is to do it how we used to do it in the AvailabilityChecker before contexts:
```
	pollStartTime := time.Now()
	for time.Since(pollStartTime) < startupTimeout {
		if checker.core.IsServiceUp(checker.toCheck, checker.dependencies) {
			return nil
		}
		logrus.Tracef("Service is not yet available; sleeping for %v before retrying...", TIME_BETWEEN_STARTUP_POLLS)
		time.Sleep(TIME_BETWEEN_STARTUP_POLLS)
	}
```

Which also lets you specify the networkAcceptanceTime in a time.Duration (which is a little friendlier)

Reply:
done!